### DISCUSSION TITLE
Repo creation

### DISCUSSION THREAD

**Discussion by ZanisO** [argoproj/argo-cd#9757] [Q&A]
Hi,

I'm using Azure DevOps (private repository) to store manifests used by Argo to sync the applications. For Argo to have visibility of the manifests, I have created Repo using [this guide](https://argo-cd.readthedocs.io/en/stable/user-guide/private-repositories/).

I was wondering if the only way to add a Repo, in Argo, is by creating Secret directly (via `kubectl apply`) or indirectly (via Argo CLI or UI)? Is it possible to create repository in Argo directly using CSI driver and Azure Key Vault?

**Comment by jannfis** [comment]
Argo CD doesn't care who or which controller created the secret, as long as it has the required label (`argocd.argoproj.io/secret-type: repository`) set and has well-formed  configuration data in it.

So if you are able to configure your CSI driver to fullfil these requirements, you should be fine.
